packages/console/src/ds-components/Select/index.tsx

```diff
@@ -108,7 +164,31 @@ function Select<T extends string>({
           setIsOpen(false);
         }}
       >
-        {options.map(({ value, title }) => (
+        {isSearchEnabled && isOpen && (
+          <div
+            className={styles.searchInputContainer}
+            style={{
+              position: 'fixed',
+              ...getSearchInputContainerStyles(),
+            }}
+          >
+            <SearchIcon className={styles.search} />
+            <input
+              ref={(input) => input?.focus()}
+              value={searchInputValue}
+              role="searchbox"
+              autoComplete="off"
+              placeholder={t('general.type_to_search')}
+              onChange={(event) => {
+                setSearchInputValue(event.target.value);
+              }}
+              onClick={(event) => {
+                event.stopPropagation();
+              }}
+            />
+          </div>
```

Comment:
nit-picking: Would it be easier to use our `Input` component instead of writing it from scratch? Seems some of the CSS styles can be saved.

Reply:
@charIeszhao I wanted to use the outline and border of the select, so I didn't use the Input component.